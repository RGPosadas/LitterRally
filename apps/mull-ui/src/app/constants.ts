export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
