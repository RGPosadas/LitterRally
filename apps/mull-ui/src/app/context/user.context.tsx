import React from 'react';

const UserContext = React.createContext<{
  userId: number;
  setUserId: React.Dispatch<React.SetStateAction<number>>;
  accessToken: string;
  setAccessToken: React.Dispatch<React.SetStateAction<string>>;
}>(null);

export const UserProvider = UserContext.Provider;
export const UserConsumer = UserContext.Consumer;

export default UserContext;
