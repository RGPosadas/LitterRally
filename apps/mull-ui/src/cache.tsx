import { InMemoryCache, ReactiveVar, makeVar } from '@apollo/client';

export const cache: InMemoryCache = new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        login: {
          read() {
            return isLoggedIn();
          },
        },
      },
    },
  },
});

const loginInitialValue: boolean = false;

export const isLoggedIn: ReactiveVar<boolean> = makeVar<boolean>(loginInitialValue);
