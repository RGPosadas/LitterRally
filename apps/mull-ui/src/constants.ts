import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
