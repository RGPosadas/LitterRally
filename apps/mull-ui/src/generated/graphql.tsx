import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** A date-time string at UTC, such as 2019-12-03T09:54:33Z, compliant with the date-time format. */
  DateTime: any;
  /** The `Upload` scalar type represents a file upload. */
  Upload: any;
};

export type CreateEventInput = {
  description: Scalars['String'];
  endDate: Scalars['DateTime'];
  image: MediaInput;
  restriction: Scalars['Int'];
  startDate: Scalars['DateTime'];
  title: Scalars['String'];
};

export type CreateUserInput = {
  dob?: Maybe<Scalars['DateTime']>;
  email: Scalars['String'];
  name: Scalars['String'];
  password?: Maybe<Scalars['String']>;
  registrationMethod: RegistrationMethod;
};

export type Event = {
  __typename?: 'Event';
  description: Scalars['String'];
  endDate: Scalars['DateTime'];
  id: Scalars['ID'];
  image?: Maybe<Media>;
  restriction: Scalars['Float'];
  startDate: Scalars['DateTime'];
  title: Scalars['String'];
};

export type LoginInput = {
  email: Scalars['String'];
  password: Scalars['String'];
};

export type LoginResult = {
  __typename?: 'LoginResult';
  accessToken: Scalars['String'];
};

export type Media = {
  __typename?: 'Media';
  id: Scalars['Float'];
  mediaType: Scalars['String'];
};

export type MediaInput = {
  id: Scalars['Float'];
  mediaType: Scalars['String'];
};

export type Mutation = {
  __typename?: 'Mutation';
  addParticipantToEvent: Scalars['Boolean'];
  createEvent: Event;
  createUser: User;
  deleteEvent: Event;
  deleteUser: User;
  login: LoginResult;
  removeParticipantFromEvent: Scalars['Boolean'];
  updateEvent: Event;
  updateUser: User;
  uploadFile: Media;
};

export type MutationAddParticipantToEventArgs = {
  eventId: Scalars['Float'];
  userId: Scalars['Float'];
};

export type MutationCreateEventArgs = {
  createEventInput: CreateEventInput;
};

export type MutationCreateUserArgs = {
  createUserInput: CreateUserInput;
};

export type MutationDeleteEventArgs = {
  id: Scalars['Int'];
};

export type MutationDeleteUserArgs = {
  id: Scalars['Int'];
};

export type MutationLoginArgs = {
  loginInput: LoginInput;
};

export type MutationRemoveParticipantFromEventArgs = {
  eventId: Scalars['Float'];
  userId: Scalars['Float'];
};

export type MutationUpdateEventArgs = {
  updateEventInput: UpdateEventInput;
};

export type MutationUpdateUserArgs = {
  updateUserInput: UpdateUserInput;
};

export type MutationUploadFileArgs = {
  file: Scalars['Upload'];
};

export type Query = {
  __typename?: 'Query';
  coHostEvents: Array<Event>;
  discoverEvents: Array<Event>;
  event: Event;
  events: Array<Event>;
  hostEvents: Array<Event>;
  participatingEvents: Array<Event>;
  user: User;
  users: Array<User>;
};

export type QueryCoHostEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryDiscoverEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryEventArgs = {
  id: Scalars['Int'];
};

export type QueryHostEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryParticipatingEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryUserArgs = {
  id: Scalars['Int'];
};

export enum RegistrationMethod {
  Facebook = 'FACEBOOK',
  Google = 'GOOGLE',
  Local = 'LOCAL',
  Twitter = 'TWITTER',
}

export type UpdateEventInput = {
  description?: Maybe<Scalars['String']>;
  endDate?: Maybe<Scalars['DateTime']>;
  id: Scalars['ID'];
  restriction?: Maybe<Scalars['Int']>;
  startDate?: Maybe<Scalars['DateTime']>;
  title?: Maybe<Scalars['String']>;
};

export type UpdateUserInput = {
  dob?: Maybe<Scalars['DateTime']>;
  email?: Maybe<Scalars['String']>;
  id: Scalars['ID'];
  name?: Maybe<Scalars['String']>;
};

export type User = {
  __typename?: 'User';
  description: Scalars['String'];
  dob: Scalars['DateTime'];
  email: Scalars['String'];
  friends: Array<User>;
  id: Scalars['ID'];
  name: Scalars['String'];
  password?: Maybe<Scalars['String']>;
  registrationMethod: RegistrationMethod;
  timezone: Scalars['String'];
};

export type CreateEventMutationVariables = Exact<{
  createEventInput: CreateEventInput;
}>;

export type CreateEventMutation = { __typename?: 'Mutation' } & {
  createEvent: { __typename?: 'Event' } & Pick<Event, 'id'>;
};

export type UploadFileMutationVariables = Exact<{
  file: Scalars['Upload'];
}>;

export type UploadFileMutation = { __typename?: 'Mutation' } & {
  uploadFile: { __typename?: 'Media' } & Pick<Media, 'id' | 'mediaType'>;
};

export type FindSpecificEventQueryVariables = Exact<{
  eventId: Scalars['Int'];
}>;

export type FindSpecificEventQuery = { __typename?: 'Query' } & {
  event: { __typename?: 'Event' } & Pick<
    Event,
    'id' | 'title' | 'description' | 'startDate' | 'endDate' | 'restriction'
  >;
};

export type DiscoverEventsQueryVariables = Exact<{
  discoverEventsUserId: Scalars['Int'];
}>;

export type DiscoverEventsQuery = { __typename?: 'Query' } & {
  discoverEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
};

export type GetUsersEventsQueryVariables = Exact<{
  UserId: Scalars['Int'];
}>;

export type GetUsersEventsQuery = { __typename?: 'Query' } & {
  coHostEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
  hostEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
};

export type GetParticipatingEventsQueryVariables = Exact<{
  participatingEventsUserId: Scalars['Int'];
}>;

export type GetParticipatingEventsQuery = { __typename?: 'Query' } & {
  participatingEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'endDate' | 'description' | 'startDate' | 'title'>
  >;
};

export const CreateEventDocument = gql`
  mutation CreateEvent($createEventInput: CreateEventInput!) {
    createEvent(createEventInput: $createEventInput) {
      id
    }
  }
`;
export type CreateEventMutationFn = Apollo.MutationFunction<
  CreateEventMutation,
  CreateEventMutationVariables
>;

/**
 * __useCreateEventMutation__
 *
 * To run a mutation, you first call `useCreateEventMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useCreateEventMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [createEventMutation, { data, loading, error }] = useCreateEventMutation({
 *   variables: {
 *      createEventInput: // value for 'createEventInput'
 *   },
 * });
 */
export function useCreateEventMutation(
  baseOptions?: Apollo.MutationHookOptions<CreateEventMutation, CreateEventMutationVariables>
) {
  return Apollo.useMutation<CreateEventMutation, CreateEventMutationVariables>(
    CreateEventDocument,
    baseOptions
  );
}
export type CreateEventMutationHookResult = ReturnType<typeof useCreateEventMutation>;
export type CreateEventMutationResult = Apollo.MutationResult<CreateEventMutation>;
export type CreateEventMutationOptions = Apollo.BaseMutationOptions<
  CreateEventMutation,
  CreateEventMutationVariables
>;
export const UploadFileDocument = gql`
  mutation UploadFile($file: Upload!) {
    uploadFile(file: $file) {
      id
      mediaType
    }
  }
`;
export type UploadFileMutationFn = Apollo.MutationFunction<
  UploadFileMutation,
  UploadFileMutationVariables
>;

/**
 * __useUploadFileMutation__
 *
 * To run a mutation, you first call `useUploadFileMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useUploadFileMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [uploadFileMutation, { data, loading, error }] = useUploadFileMutation({
 *   variables: {
 *      file: // value for 'file'
 *   },
 * });
 */
export function useUploadFileMutation(
  baseOptions?: Apollo.MutationHookOptions<UploadFileMutation, UploadFileMutationVariables>
) {
  return Apollo.useMutation<UploadFileMutation, UploadFileMutationVariables>(
    UploadFileDocument,
    baseOptions
  );
}
export type UploadFileMutationHookResult = ReturnType<typeof useUploadFileMutation>;
export type UploadFileMutationResult = Apollo.MutationResult<UploadFileMutation>;
export type UploadFileMutationOptions = Apollo.BaseMutationOptions<
  UploadFileMutation,
  UploadFileMutationVariables
>;
export const FindSpecificEventDocument = gql`
  query FindSpecificEvent($eventId: Int!) {
    event(id: $eventId) {
      id
      title
      description
      startDate
      endDate
      restriction
    }
  }
`;

/**
 * __useFindSpecificEventQuery__
 *
 * To run a query within a React component, call `useFindSpecificEventQuery` and pass it any options that fit your needs.
 * When your component renders, `useFindSpecificEventQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useFindSpecificEventQuery({
 *   variables: {
 *      eventId: // value for 'eventId'
 *   },
 * });
 */
export function useFindSpecificEventQuery(
  baseOptions: Apollo.QueryHookOptions<FindSpecificEventQuery, FindSpecificEventQueryVariables>
) {
  return Apollo.useQuery<FindSpecificEventQuery, FindSpecificEventQueryVariables>(
    FindSpecificEventDocument,
    baseOptions
  );
}
export function useFindSpecificEventLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<FindSpecificEventQuery, FindSpecificEventQueryVariables>
) {
  return Apollo.useLazyQuery<FindSpecificEventQuery, FindSpecificEventQueryVariables>(
    FindSpecificEventDocument,
    baseOptions
  );
}
export type FindSpecificEventQueryHookResult = ReturnType<typeof useFindSpecificEventQuery>;
export type FindSpecificEventLazyQueryHookResult = ReturnType<typeof useFindSpecificEventLazyQuery>;
export type FindSpecificEventQueryResult = Apollo.QueryResult<
  FindSpecificEventQuery,
  FindSpecificEventQueryVariables
>;
export const DiscoverEventsDocument = gql`
  query DiscoverEvents($discoverEventsUserId: Int!) {
    discoverEvents(userId: $discoverEventsUserId) {
      id
      title
      description
      startDate
      endDate
    }
  }
`;

/**
 * __useDiscoverEventsQuery__
 *
 * To run a query within a React component, call `useDiscoverEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useDiscoverEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useDiscoverEventsQuery({
 *   variables: {
 *      discoverEventsUserId: // value for 'discoverEventsUserId'
 *   },
 * });
 */
export function useDiscoverEventsQuery(
  baseOptions: Apollo.QueryHookOptions<DiscoverEventsQuery, DiscoverEventsQueryVariables>
) {
  return Apollo.useQuery<DiscoverEventsQuery, DiscoverEventsQueryVariables>(
    DiscoverEventsDocument,
    baseOptions
  );
}
export function useDiscoverEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<DiscoverEventsQuery, DiscoverEventsQueryVariables>
) {
  return Apollo.useLazyQuery<DiscoverEventsQuery, DiscoverEventsQueryVariables>(
    DiscoverEventsDocument,
    baseOptions
  );
}
export type DiscoverEventsQueryHookResult = ReturnType<typeof useDiscoverEventsQuery>;
export type DiscoverEventsLazyQueryHookResult = ReturnType<typeof useDiscoverEventsLazyQuery>;
export type DiscoverEventsQueryResult = Apollo.QueryResult<
  DiscoverEventsQuery,
  DiscoverEventsQueryVariables
>;
export const GetUsersEventsDocument = gql`
  query GetUsersEvents($UserId: Int!) {
    coHostEvents(userId: $UserId) {
      id
      title
      description
      startDate
      endDate
    }
    hostEvents(userId: $UserId) {
      id
      title
      description
      startDate
      endDate
    }
  }
`;

/**
 * __useGetUsersEventsQuery__
 *
 * To run a query within a React component, call `useGetUsersEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetUsersEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetUsersEventsQuery({
 *   variables: {
 *      UserId: // value for 'UserId'
 *   },
 * });
 */
export function useGetUsersEventsQuery(
  baseOptions: Apollo.QueryHookOptions<GetUsersEventsQuery, GetUsersEventsQueryVariables>
) {
  return Apollo.useQuery<GetUsersEventsQuery, GetUsersEventsQueryVariables>(
    GetUsersEventsDocument,
    baseOptions
  );
}
export function useGetUsersEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<GetUsersEventsQuery, GetUsersEventsQueryVariables>
) {
  return Apollo.useLazyQuery<GetUsersEventsQuery, GetUsersEventsQueryVariables>(
    GetUsersEventsDocument,
    baseOptions
  );
}
export type GetUsersEventsQueryHookResult = ReturnType<typeof useGetUsersEventsQuery>;
export type GetUsersEventsLazyQueryHookResult = ReturnType<typeof useGetUsersEventsLazyQuery>;
export type GetUsersEventsQueryResult = Apollo.QueryResult<
  GetUsersEventsQuery,
  GetUsersEventsQueryVariables
>;
export const GetParticipatingEventsDocument = gql`
  query GetParticipatingEvents($participatingEventsUserId: Int!) {
    participatingEvents(userId: $participatingEventsUserId) {
      id
      endDate
      description
      startDate
      title
    }
  }
`;

/**
 * __useGetParticipatingEventsQuery__
 *
 * To run a query within a React component, call `useGetParticipatingEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetParticipatingEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetParticipatingEventsQuery({
 *   variables: {
 *      participatingEventsUserId: // value for 'participatingEventsUserId'
 *   },
 * });
 */
export function useGetParticipatingEventsQuery(
  baseOptions: Apollo.QueryHookOptions<
    GetParticipatingEventsQuery,
    GetParticipatingEventsQueryVariables
  >
) {
  return Apollo.useQuery<GetParticipatingEventsQuery, GetParticipatingEventsQueryVariables>(
    GetParticipatingEventsDocument,
    baseOptions
  );
}
export function useGetParticipatingEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<
    GetParticipatingEventsQuery,
    GetParticipatingEventsQueryVariables
  >
) {
  return Apollo.useLazyQuery<GetParticipatingEventsQuery, GetParticipatingEventsQueryVariables>(
    GetParticipatingEventsDocument,
    baseOptions
  );
}
export type GetParticipatingEventsQueryHookResult = ReturnType<
  typeof useGetParticipatingEventsQuery
>;
export type GetParticipatingEventsLazyQueryHookResult = ReturnType<
  typeof useGetParticipatingEventsLazyQuery
>;
export type GetParticipatingEventsQueryResult = Apollo.QueryResult<
  GetParticipatingEventsQuery,
  GetParticipatingEventsQueryVariables
>;
