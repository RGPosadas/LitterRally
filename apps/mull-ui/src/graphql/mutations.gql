# Creating Events
mutation CreateEvent($event: CreateEventInput!) {
  createEvent(event: $event) {
    id
  }
}

# Uploading media files
mutation UploadFile($file: Upload!) {
  uploadFile(file: $file) {
    id
    mediaType
  }
}

# Register a user
mutation CreateUser($user: CreateUserInput!) {
  createUser(user: $user) {
    id
  }
}

# Login a user
mutation Login($loginInput: LoginInput!) {
  login(loginInput: $loginInput) {
    accessToken
  }
# User joins an event
mutation JoinEvent($eventId: Int!, $userId: Int!) {
  addParticipantToEvent(eventId: $eventId, userId: $userId)
}

# User leaves an event
mutation LeaveEvent($eventId: Int!, $userId: Int!) {
  removeParticipantFromEvent(eventId: $eventId, userId: $userId)
}
