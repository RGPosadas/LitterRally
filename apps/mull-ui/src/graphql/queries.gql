fragment EventPageContent on Event {
  id
  title
  description
  startDate
  endDate
  restriction
  location {
    title
  }
  image {
    id
    mediaType
  }
  host {
    id
    name
  }
}

fragment EventCardContent on Event {
  id
  title
  restriction
  startDate
  location {
    title
  }
  image {
    id
    mediaType
  }
}

query DiscoverEvents {
  discoverEvents {
    ...EventCardContent
  }
}

# Gets all related events to a user
query OwnedEvents {
  coHostEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
  hostEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
}

# Get all the events that a user is participating in
query ParticipantEvents {
  participantEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
}

# Find events by Id and finds participating events of user
query EventPage($eventId: Int!) {
  isParticipant(eventId: $eventId)

  event(id: $eventId) {
    ...EventPageContent
  }
}

query User {
  user {
    id
    name
    description
    avatar {
      id
    }
  }
}

query UserProfile {
  user {
    name
    description
    joinDate
    avatar {
      id
    }
  }

  friendCount
  hostingCount
  portfolioCount
}
