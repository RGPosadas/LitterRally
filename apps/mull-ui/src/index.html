<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Müll</title>
    <base href="/" />

    <link rel="icon" type="image/x-icon" href="favicon.png" />
    <link rel="manifest" href="manifest.json" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="description" content="Müll: Help cleanup the world!" />
    <meta name="robots" content="index, follow" />
    <meta name="theme-color" content="#27b09a" />
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="default" />
    <meta name="apple-mobile-web-app-title" content="my-app" />
    <link rel="apple-touch-icon" href="assets/icons/apple-touch-icon.png" />
  </head>
  <body>
    <div id="root"></div>
    <noscript>
      <div style="font-size: 20px; display: block">
        <div>Please enable JavaScript to continue using this application.</div>
        <div>
          <a href="https://www.enable-javascript.com">https://www.enable-javascript.com</a>
        </div>
      </div>
    </noscript>
  </body>
</html>
