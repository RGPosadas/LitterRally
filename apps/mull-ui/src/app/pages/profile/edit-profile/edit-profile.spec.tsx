import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import '@testing-library/jest-dom/extend-expect';
import { act, fireEvent, render, waitFor } from '@testing-library/react';
import user from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer, { act as actRenderer } from 'react-test-renderer';
import { ROUTES } from '../../../../../src/constants';
import { UserDocument, UserQuery } from '../../../../generated/graphql';
import { UserProvider } from '../../../context/user.context';
import EditProfile from './edit-profile';

describe('EditProfile', () => {
  const renderHelper = (history, mocks: MockedResponse[]) => {
    return (
      <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Router history={history}>
            <EditProfile history={history} />
          </Router>
        </MockedProvider>
      </UserProvider>
    );
  };

  it('should render successfully', () => {
    const history = createMemoryHistory();
    history.push(ROUTES.PROFILE.EDIT);

    const { baseElement } = render(renderHelper(history, null));
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserDocument,
        },
        result: {
          data: {
            user: {
              id: 1,
              name: 'Jane Doe',
              description: 'a description',
              avatar: null,
            },
          } as UserQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.PROFILE.DISPLAY);
    await actRenderer(async () => {
      const tree = renderer.create(renderHelper(history, mocks));
      await new Promise((resolve) => setTimeout(resolve, 100));
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  it('should have validation errors', async () => {
    const history = createMemoryHistory();
    history.push(ROUTES.PROFILE.EDIT);
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserDocument,
        },
        result: {
          data: {
            user: {
              id: 1,
              name: '',
              description: 'a description',
              avatar: null,
            },
          } as UserQuery,
        },
      },
    ];

    await act(async () => {
      const utils = render(renderHelper(history, mocks));
      await new Promise((resolve) => setTimeout(resolve, 100));
      const saveButton = utils.container.querySelector('button[data-testid="mull-button"]');

      await waitFor(() => {
        fireEvent.click(saveButton);
      });

      const error = utils.container.querySelector('span[class="error-message"]');
      expect(error).toHaveTextContent('Display name is required');
    });
  });

  it('should submit the edit profile form', async () => {
    const history = createMemoryHistory();
    history.push(ROUTES.PROFILE.EDIT);
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserDocument,
        },
        result: {
          data: {
            user: {
              id: 1,
              name: 'Jane Doe',
              description: 'a description',
              avatar: null,
            },
          } as UserQuery,
        },
      },
    ];

    await act(async () => {
      const utils = render(renderHelper(history, mocks));
      await new Promise((resolve) => setTimeout(resolve, 100));

      window.URL.createObjectURL = jest.fn();
      const file = new File(['zoro'], 'zoro.png', { type: 'image/png' });
      const imageInput = utils.getByTestId('file') as HTMLInputElement;
      await waitFor(() => {
        user.upload(imageInput, file);
      });
      let input = utils.getByLabelText('Display Name');
      fireEvent.change(input, { target: { value: 'Bob' } });
      input = utils.getByLabelText('Description');
      fireEvent.change(input, { target: { value: 'bob the builder' } });

      const submitButton = utils.container.querySelector('button[class="mull-button save-button"]');

      await waitFor(() => {
        fireEvent.click(submitButton);
      });
    });
  });
});
