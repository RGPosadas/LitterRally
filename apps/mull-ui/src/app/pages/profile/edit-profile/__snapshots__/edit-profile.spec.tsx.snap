// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile should match snapshot 1`] = `
<div
  className="page-container"
>
  <button
    className="mull-back-button  "
    data-testid="mull-back-button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-left fa-w-10 mull-back-button-icon"
      data-icon="chevron-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      className="mull-back-button-text"
    >
      Profile
    </div>
  </button>
  <form
    className="edit-profile-container"
    onSubmit={[Function]}
  >
    <p
      className="edit-header"
    >
      Edit Profile
    </p>
    <div
      className="custom-file-upload-container"
    >
      <label
        className="image-uploaded"
        htmlFor="imageFile"
      >
        <img
          alt="Event"
          className="custom-file-upload-image"
          src="./assets/icons/icon-192x192.png"
        />
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-pencil-alt fa-w-16 custom-file-upload-pencil-icon button"
          data-icon="pencil-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </label>
      <input
        accept="image/*"
        className="custom-file-upload-input"
        data-testid="file"
        id="imageFile"
        onChange={[Function]}
        type="file"
      />
    </div>
    <div
      className="custom-text-input-container "
    >
      <label
        className="custom-text-input-label"
        htmlFor="displayName"
      >
        Display Name
      </label>
      <div
        className="custom-text-input-sub-container"
      >
        <input
          className="custom-text-input input-border "
          data-testid="custom-text-input"
          id="displayName"
          name="displayName"
          onChange={[Function]}
          type="text"
          value="Jane Doe"
        />
        <div
          className="custom-text-input-icon"
        />
      </div>
    </div>
    <label
      className="description-label"
      htmlFor="description"
    >
      Description
    </label>
    <textarea
      className="description-msg "
      id="description"
      onChange={[Function]}
      rows={6}
      value="a description"
    />
    <button
      className="mull-button save-button "
      data-testid="mull-button"
      onClick={[Function]}
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
